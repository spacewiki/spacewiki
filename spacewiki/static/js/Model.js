export class Page {
  constructor(pageJSON, slug) {
    if (pageJSON) {
      this.data = pageJSON;
    } else {
      this.data = {
        latest: {
          timestamp: "Today",
          author: {
            display: "Anonymous"
          }
        }
      }
    }

    this.latestRevision = new Revision(this.data.latest || {}, this);
    this.navigation = this.data.navigation || {subpages: [], parents: [], siblings: []};
    this.softlinks = this.data.softlinks || [];
    this.slug = this.data.slug || slug;
    this.title = this.data.title || this.slug;
    this.url = '/' + this.slug
  }
}

export class Revision {
  constructor(revisionJSON, parentPage) {
    if (revisionJSON) {
      this.data = revisionJSON;
    } else {
      this.data = {
        timestamp: "Today",
        author: {
          display: "Anonymous"
        }
      }
    }

    this.body = this.data.body || "";
    this.id = this.data.id || 0;
    this.page = parentPage;
    this.timestamp = this.data.timestamp || "Today";
    this.author = new Identity(this.data.author || {});
    this.rendered = this.data.rendered || "";
  }
}

export class Identity {
  constructor(authorJSON) {
    if (authorJSON) {
      this.data = authorJSON
    } else {
      this.data = {
        display: "Anonymous"
      }
    }

    this.display = this.data.display || "Anonymous";
  }
}
