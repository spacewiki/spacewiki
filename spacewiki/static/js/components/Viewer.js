import React, { Component } from 'react';
import { Revision } from '../Model';
import Raw from './Raw';

export default class Viewer extends Component {
  render() {
    return (
      <Raw html={this.props.revision.rendered} />
    );
  }
};

Viewer.propTypes = {
  revision: React.PropTypes.instanceOf(Revision).isRequired
}

Viewer.defaultProps = {
  revision: new Revision()
}
