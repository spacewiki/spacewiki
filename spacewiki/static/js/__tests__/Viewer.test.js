import React from 'react';
import { shallow } from 'enzyme';

import { Page } from '../Model';
import Viewer from '../components/Viewer';

test('Blank page', () => {
  var p = new Page();
  expect(shallow(<Viewer revision={p.latestRevision} />).html()).toMatchSnapshot();
});

test('Empty revision', () => {
  expect(shallow(<Viewer />).html()).toMatchSnapshot();
});
