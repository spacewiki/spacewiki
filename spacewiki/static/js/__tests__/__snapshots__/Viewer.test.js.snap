exports[`test Blank page 1`] = `"<p></p>"`;

exports[`test Empty revision 1`] = `"<p></p>"`;
